rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/CatalogRequestBuilder.java

```diff
@@ -0,0 +1,128 @@
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.entity.mime.HttpMultipartMode;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.FileBody;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+
+
+public class CatalogRequestBuilder {
+
+    private final CatalogObjectAction catalogObjectAction;
+
+    public CatalogRequestBuilder(CatalogObjectAction catalogObjectAction) {
+        this.catalogObjectAction = catalogObjectAction;
+    }
+
+    public String build() {
+        String urlEncodedCommitMessage = getUrlEncodedCommitMessageOrFail();
+        StringBuilder builder = new StringBuilder(URL_CATALOG).append("/buckets/")
+                                                              .append(this.catalogObjectAction.getBucketName());
+        if (this.catalogObjectAction.isRevised()) {
+            builder.append("/resources/").append(this.catalogObjectAction.getNodeSourceName()).append("/revisions?");
+            appendParameters(builder, paramPair(COMMIT_MESSAGE_PARAM, urlEncodedCommitMessage));
+        } else {
+            builder.append("/resources?");
+            appendParameters(builder,
+                             paramPair(NAME_PARAM, this.catalogObjectAction.getNodeSourceName()),
+                             paramPair(KIND_PARAM, this.catalogObjectAction.getKind()),
+                             paramPair(COMMIT_MESSAGE_PARAM, urlEncodedCommitMessage),
+                             paramPair(OBJECT_CONTENT_TYPE_PARAM, this.catalogObjectAction.getObjectContentType()));
+        }
+        return builder.toString();
+    }
+
+    private String getUrlEncodedCommitMessageOrFail() {
+        try {
+            return URLEncoder.encode(this.catalogObjectAction.getCommitMessage(), "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw new IllegalArgumentException("Commit message cannot be encoded properly", e);
+        }
+    }
+
+    private void appendParameters(StringBuilder builder, String... paramPairs) {
+        builder.append(Arrays.stream(paramPairs).collect(Collectors.joining("&")));
+    }
+
+    public CloseableHttpResponse postNodeSourceRequestToCatalog(String sessionId, String fullUri,
+            HttpServletResponse response)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "response". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)